### ISSUE TITLE
bayesfactor_pearson return different results than correlationBF

### ISSUE THREAD

**Issue by raphaelvallat** [raphaelvallat/pingouin#39]
See https://github.com/cran/BayesFactor/blob/0a1fe0bedf62549a466c9ec5db8b8b5a0217f0c6/R/correlationBF.R

**Comment by raphaelvallat** [comment]
Current calculation are based on https://github.com/anne-urai/Tools/blob/71c30c15c2b9a0a658873502c269783b2b63a155/stats/BayesFactors/corrbf.m

**Comment by raphaelvallat** [comment]
https://github.com/richarddmorey/BayesFactor/blob/master/pkg/BayesFactor/R/correlation-JASP.R

**Comment by raphaelvallat** [comment]
The code below gives similar results to the R BayesFactor::correlationBF function: 
```
from math import log, lgamma, pi
from scipy.special import beta, hyp2f1

r, n = 0.083, 100
kappa = 1 / 3

# Compute BF10
log_hyperterm = log(hyp2f1(((n - 1) / 2), ((n - 1) / 2), ((n + 2 / kappa) / 2), r**2))
log_result = (1 - 2 / kappa) * log(2) + 0.5 * log(pi)- log(beta(1 / kappa, 1 / kappa)) + lgamma((n + 2 / kappa - 1) / 2) - lgamma((n + 2 / kappa) / 2) + log_hyperterm
np.exp(log_result)
```

**Comment by raphaelvallat** [comment]
Resolved in 5450bfce819ac3eec1737cab2e0d52d19b12a9a7
